pandas/tests/test_resample.py

```diff
@@ -2777,6 +2694,211 @@ def test_evenly_divisible_with_no_extra_bins(self):
         assert_frame_equal(result, expected)
 
 
+def assert_series_or_frame_equal(result, expected):
+    if isinstance(result, Series):
+        return assert_series_equal(result, expected)
+    if isinstance(result, DataFrame):
+        return assert_frame_equal(result, expected)
+
+
+class Base_pytest(object):
+    """
+    base class for resampling testing providing fixtures for each index type
+    - index: default index defined by _index_factory in each subclass
+    - pandas_obj: default Series and DataFrame objects
+    """
+
+    _index_fixture_start = datetime(2005, 1, 1)
+    _index_fixture_end = datetime(2005, 1, 10)
+    _index_fixture_freq = 'D'
+
+    def create_index(self, *args, **kwargs):
+        """ return the _index_factory created using the args, kwargs """
+        factory = self._index_factory()
+        return factory(*args, **kwargs)
+
+    @pytest.fixture(scope='class')
+    def index(self):
+        return self.create_index(self._index_fixture_start,
+                                 self._index_fixture_end,
+                                 freq=self._index_fixture_freq)
+
+    @pytest.fixture(params=[Series, DataFrame], scope='class')
+    def pandas_obj(self, request, index):
+        if request.param == Series:
+            return Series(np.arange(len(index)), index=index,
+                          name=self._series_fixture_name)
+        if request.param == DataFrame:
+            return DataFrame({'value': np.arange(len(index))}, index=index)
+
+
+class TestDatetimeIndex_pytest(Base_pytest):
+    _index_factory = lambda x: date_range
+    _series_fixture_name = 'dti'
+
+    # stub, put parametrized tests here
+    pass
+
+
+class TestPeriodIndex_pytest(Base_pytest):
```

Comment:
I would rather name these ``TestPeriodIndex(Base)`` and rename the original tests I think (and then we should move them in other PR's)